miio/tests/test_airconditioningcompanion.py

```diff
@@ -56,21 +57,19 @@ def __init__(self, *args, **kwargs):
         self.state = ["010500978022222102", "01020119A280222221", "2"]
         self.last_ir_played = None
 
-        self.return_values = {
-            "get_model_and_state": self._get_state,
-            "start_ir_learn": lambda x: True,
-            "end_ir_learn": lambda x: True,
-            "get_ir_learn_result": lambda x: True,
-            "send_ir_code": lambda x: self._send_input_validation(x),
-            "send_cmd": lambda x: self._send_input_validation(x),
-            "set_power": lambda x: self._set_power(x),
-        }
+        self.command_sender = DummyCommandSender(
+            {
+                "get_model_and_state": self._get_state,
+                "start_ir_learn": lambda x: True,
+                "end_ir_learn": lambda x: True,
+                "get_ir_learn_result": lambda x: True,
+                "send_ir_code": lambda x: self._send_input_validation(x),
+                "send_cmd": lambda x: self._send_input_validation(x),
+                "set_power": lambda x: self._set_power(x),
+            }
+        )
```

Comment:
Please keep the `self.return_values` for now, and simply pass it as a parameter to the implementation. This will keep the diff much more readable and there is no real need to modify these parts at the moment (imo).

Reply:
Honestly I'd prefer it that'd be done in a separate PR. This PR contains already so many lines to review, and as these are tests, not having `return_values` does not really bring much.

Reply:
Fair enough. I'm initialising the `DummyProtocol` in `DummyDevice` now and passing in `self` so it can access `return_values`. That way the diff is indeed smaller.

Had to add call to the parent constructor for some dummy devices in tests and add inheritance from `DummyDevice`, but that should faily uncontroversial :)